django/middleware/locale.py

```diff
@@ -35,7 +35,12 @@ def process_response(self, request, response):
         urlconf = getattr(request, 'urlconf', settings.ROOT_URLCONF)
         i18n_patterns_used, prefixed_default_language = is_language_prefix_patterns_used(urlconf)
 
-        if response.status_code == 404 and not language_from_path and i18n_patterns_used:
+        if (response.status_code == 404 and
+                not language_from_path and
+                i18n_patterns_used and
+                (prefixed_default_language or language != settings.LANGUAGE_CODE)):
+            # Maybe the language code is missing in the URL, try to add the
```

Comment:
suggested:
```
# Maybe the language code is missing in the URL? Try adding the
# language prefix and redirecting to that URL.
```
Would be nice if the comment also explained the new "prefixed_default_language or language != settings.LANGUAGE_CODE" condition.